### ISSUE TITLE
gsi crashes when importing R7RS library without 'begin'

### ISSUE THREAD

**Issue by daviwil** [gambit/gambit#613]
At some point in the last month some code I wrote has started to crash `gsi` at startup when modules are being loaded.  I tracked it down to two modules, one which just re-exports procedures from other modules (no `begin` declaration) and one with only a single macro defined inside of the `begin`.

I've tried to look into the `_define-library-expand` code but I'm not familiar enough with the area to make quick progress. @feeley If you have any pointers on where I might find the issue I'd be happy to try fixing it myself.

Here's an example program which reproduces the issue:

**main.scm**

```scheme
(import (empty) ; Comment out this import to see crash on macro-only too
        (macro-only))

(println "Never makes it here")
```

**empty.scm**

```scheme
(define-library (empty)
  (import (gambit))
  (export list))
```

**macro-only.scm**
```scheme
(define-library (macro-only)
  (import (gambit))
  (begin
    (define-macro (useless-print thing)
      `(println ,thing))))
```

**Run command**

```
gsi . main.scm
```

**Comment by feeley** [comment]
That's due to a bug introduced in 98373cf53ca8d77b90e6559afe0da8032121ba07.  Now fixed by a3a2f895c36d567ba70f8720371772c5301771e3.

**Comment by daviwil** [comment]
The fix worked for me, thanks @feeley!
